Modal: add padding to container and margin top to children container

diegofontenelle opened the pull request:
# Modal spacing update

- Add 16px padding to container (x axis)
- Add 24px padding to Header container (y axis)
- Add 16px margin top to children container 
- Add 16px padding to footer container (y axis)

Proposed:
![image](https://user-images.githubusercontent.com/10033075/98745653-b8597580-2392-11eb-8d22-8e963cc78256.png)

Implemented:
![image](https://user-images.githubusercontent.com/10033075/98856875-fdd27d00-243c-11eb-9347-b366a12b4aba.png)
![image](https://user-images.githubusercontent.com/10033075/98856890-0460f480-243d-11eb-9787-169068ba6a7c.png)



On smaller windows: 
![image](https://user-images.githubusercontent.com/10033075/98856910-0a56d580-243d-11eb-9e7e-d0f96cac89ef.png)

Sheet: 
![image](https://user-images.githubusercontent.com/10033075/98856927-12167a00-243d-11eb-8b95-a929ae832437.png)




## Test Plan

Manual run

christianvuerings commented:
@diegofontenelle thanks for the PR! @anniemochi could you verify from the design side?

Couple of notes:
* We should also update `Sheet` which uses similar padding
* The current change breaks scrolling (happens on windows with smaller heights). So the padding still needs to be on the outward component, it can just be less than before.
![image](https://user-images.githubusercontent.com/127199/98749395-70c1f200-2370-11eb-8030-5aa41d77772e.png)

diegofontenelle commented:
> @diegofontenelle thanks for the PR! @anniemochi could you verify from the design side?
> 
> Couple of notes:
> 
> * We should also update `Sheet` which uses similar padding
> * The current change breaks scrolling (happens on windows with smaller heights). So the padding still needs to be on the outward component, it can just be less than before.
>   ![image](https://user-images.githubusercontent.com/127199/98749395-70c1f200-2370-11eb-8030-5aa41d77772e.png)

Hi Chris! Thanks for reviewing, I have updated the PR and added new images showing how it looks right now. Let me know if this is good or you meant something else

christianvuerings commented:
Discussed with @anniemochi and @diegofontenelle: We will communicate this change internally first before merging

diegofontenelle commented:
Making small changes to the padding of the children container on both components after reviewing the designs with @anniemochi

diegofontenelle commented:
After further investigating the modal spacing adjustments we were proposing in relation to modal/sheet examples from the Monetization org, the Gestalt team has determined that we should not make those spacing changes after all (because all of the Monetization modal/sheet titles and content are left-aligned, changing the header/footer padding creates awkward alignment issues which would take a lot more effort to address).
